src/System.Runtime.Loader/tests/CollectibleAssemblyLoadContextTest.cs

```diff
@@ -0,0 +1,554 @@
+            }
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            public void CheckNotUnloaded()
+            {
+                // None of the AssemblyLoadContexts should be unloaded
+                _checker.GcAndCheck(0);
+            }
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            public void CheckContextUnloaded1()
+            {
+                // The AssemblyLoadContext should now be unloaded
+                _checker.GcAndCheck();
+            }
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            public void CheckContextUnloaded2()
+            {
+                // The AssemblyLoadContext should now be unloaded
+                _checker.GcAndCheck(1);
+            }
+        }
+
+        [Fact]
+        public static void Unload_TwoCollectibleWithOneAssemblyAndOneInstanceReferencingAnother()
+        {
+            // We create 2 collectible ALC, load one assembly in each, create one instance in each, reference one instance from ALC1 to ALC2
+            // unload ALC2 -> check that instance 2 is there
+            // unload ALC1 -> we should receive 2 unload and instance should be gone
+            // We should receive the unloading event
+
+            var test = new TwoCollectibleWithOneAssemblyAndOneInstanceReferencingAnotherTest();
+            test.CreateContextAndLoadAssembly(0);
+            test.CreateContextAndLoadAssembly(1);
+
+            test.Execute();
+
+            test.UnloadAndClearContext(1);
+
+            test.CheckNotUnloaded();
+
+            test.ReleaseInstance();
+
+            test.CheckContextUnloaded2();
+
+            test.UnloadAndClearContext(0);
+            test.CheckContextUnloaded1();
+        }
+
+        [Fact]
+        public static void Unsupported_ThreadStaticAndFixedAddressValueType()
+        {
+            var asmName = new AssemblyName(TestAssemblyNotSupported);
+            var alc = new ResourceAssemblyLoadContext(true) { LoadBy = LoadBy.Path };
+            var asm = alc.LoadFromAssemblyName(asmName);
+
+            Assert.NotNull(asm);
+
+            var exception = Assert.Throws<ReflectionTypeLoadException>(() => asm.DefinedTypes);
+            Assert.Equal(2, exception.LoaderExceptions.Length);
+            Assert.True(exception.LoaderExceptions.Any(exp => exp.Message.Contains("Collectible type 'System.Runtime.Loader.Tests.TestClassNotSupported_ThreadStatic' may not have Thread or Context static members")));
+            Assert.True(exception.LoaderExceptions.Any(exp => exp.Message.Contains("Collectible type 'System.Runtime.Loader.Tests.TestClassNotSupported_FixedAddressValueType' has unsupported FixedAddressValueTypeAttribute applied to a field")));
+        }
+
+        class DelegateMarshallingTest : TestBase
+        {
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            public void Execute()
+            {
+                var exception = Assert.Throws<TargetInvocationException>(() =>
```

Comment:
You can use `BindingFlags.DoNotWrapExceptions` and then use `Assert.Throws` on the actual exception you are interested in.